python-package/lightgbm/dask.py

```diff
@@ -886,6 +888,75 @@ def _predict(
             **kwargs
         ).values
     elif isinstance(data, dask_Array):
+        # for multi-class classification with sparse matrices, pred_contrib predictions
+        # are returned as a list of sparse matrices (one per class)
+        num_classes = getattr(model, "n_classes_", -1)
```

Comment:
Oh I did `getattr()` because I thought that `.n_classes_` and `._n_classes` would not be defined if using `LGBMRanker` or `LGBMRegressor`.

But I can see now that I was wrong about that and that `self._n_classes` is initialized to `None` in the constructor of `LGBMModel`.

Thanks for pointing it out!